### PR TITLE
Custom Fonts Addon

### PR DESCRIPTION
**Pull request by ErrorGamer2000**
**Changes**

Added new addon that allows you to customize the fonts on the scratch website... including on blocks.

### CODE REVIEW

File: addons/custom-fonts/userscript.js [hunk 1]
```diff
@@ -15,671 +15,7 @@ export default async function ({ addon, global, console }) {
      * the License.
      */
     /* Web Font Loader v1.6.26 - (c) Adobe Systems, Google. License: Apache 2.0 */
-    !(function () {
```
Comment (ErrorGamer2000) [inline addons/custom-fonts/userscript.js]: Not deleted, just minified.

File: addons/custom-fonts/userscript.js [hunk 2]
```diff
@@ -15,671 +15,7 @@ export default async function ({ addon, global, console }) {
...
-            load: i(it.load, it),
-        };
-        "function" == typeof define && define.amd
-            ? define(function () {
-                  return et;
-              })
-            : "undefined" != typeof module && module.exports
-            ? (module.exports = et)
-            : ((window.WebFont = et), window.WebFontConfig && it.load(window.WebFontConfig));
-    })();
```
Comment (ErrorGamer2000) [inline addons/custom-fonts/userscript.js]: Not deleted, just minified to make the rest of the code easier to find.

**Comment by ErrorGamer2000** [comment]
No, there aren't yet. I am also adding bold and italic options for smaller fonts. And yes, an extension like this would work for any site... but people don't necessarily want it for other sites as well.

**Comment by ErrorGamer2000** [comment]
Added bold and italic options.

File: addons/custom-fonts/userscript.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
...kit.config.fn){n=i.Typekit.config.fn;for(var e=[],o=0;o<n.length;o+=2)for(var a=n[o],s=n[o+1],r=0;r<s.length;r++)e.push(new y(a,s[r]));try{i.Typekit.load({events:!1,classes:!1,async:!0})}catch(t){}t(e)}},2e3):t([])},nt.prototype.load=function(t){var n=this.f.id,i=this.c.m,e=this;n?(i.__webfontfontdeckmodule__||(i.__webfontfontdeckmodule__={}),i.__webfontfontdeckmodule__[n]=function(n,i){for(var o=0,a=i.fonts.length;o<a;++o){var s=i.fonts[o];e.a.push(new y(s.name,_("font-weight:"+s.weight+";font-style:"+s.style)))}t(e.a)},p(this.c,l(this.c)+(this.f.api||"//f.fontdeck.com/s/css/js/")+function(t){return t.m.location.hostname||t.a.location.hostname}(this.c)+"/"+n+".js",function(n){n&&t([])})):t([])};var it=new H(window);it.a.c.custom=function(t,n){return new G(n,t)},it.a.c.fontdeck=function(t,n){return new nt(n,t)},it.a.c.monotype=function(t,n){return new z(n,t)},it.a.c.typekit=function(t,n){return new tt(n,t)},it.a.c.google=function(t,n){return new Y(n,t)};var et={load:i(it.load,it)};"function"==typeof define&&define.amd?define(function(){return et}):"undefined"!=typeof module&&module.exports?module.exports=et:(window.WebFont=et,window.WebFontConfig&&it.load(window.WebFontConfig))}();
```
Comment (RedGuy12) [inline addons/custom-fonts/userscript.js:18]: move to a library

File: addons/custom-fonts/addon.json [hunk 4]
```diff
@@ -0,0 +1,38 @@
+{
+    "name": "Custom Fonts",
+    "description": "Change the fonts on Scratch! Uses google's WebFont loader. Only fonts avaliable on google fonts (and from google) can be used. Find the font you like then paste the name into the box!",
```
Comment (RedGuy12) [inline addons/custom-fonts/addon.json:3]: "Uses google's WebFont loader. Only fonts avaliable on google fonts (and from google) can be used. Find the font you like then paste the name into the box!" is a little long...

Suggested new description: "Customize the font. Pick any font from https://fonts.google.com/ and type it below."

**Review by RedGuy12 — CHANGES_REQUESTED** [review]

File: addons/custom-fonts/userscript.js (hunk 3 above)
Comment (ErrorGamer2000) [reply addons/custom-fonts/userscript.js:18]: okay

**Comment by TheColaber** [comment]
@ErrorGamer2000 Like W_L said we cannot accept this because, many other extensions (https://chrome.google.com/webstore/detail/font-customizer/fhoncdcfffaplldeekejehabjefniidh?hl=en for example) are built just for doing this. 
> however an addon like this would work on any site
So I think it's better to just download a separate extension.

**Pull request closed without merging**